packages/esm-patient-appointments-app/src/appointments/appointments-form.test.tsx

```diff
@@ -1,175 +1,209 @@
-import { mockLocations } from '../../../../__mocks__/location.mock';
-import { useLocations, useSessionUser, detach } from '@openmrs/esm-framework';
+import { mockLocations, mockLocationsDataResponse } from '../../../../__mocks__/location.mock';
+import { detach, openmrsFetch, showNotification, showToast } from '@openmrs/esm-framework';
 import { mockSessionDataResponse } from '../../../../__mocks__/session.mock';
-import { mockUseAppointmentServiceData } from '../../../../__mocks__/appointments.mock';
-import { createAppointment } from './appointments.resource';
-import useAppointmentService from '../hooks/useAppointmentService';
+import { mockAppointmentsData, mockUseAppointmentServiceData } from '../../../../__mocks__/appointments.mock';
+import { swrRender, waitForLoadingToFinish } from '../../../../tools/test-helpers';
+import { appointmentsSearchUrl, createAppointment } from './appointments.resource';
+import AppointmentForm from './appointments-form.component';
 
-const mockCloseWorkspace = jest.fn();
-const mockUseLocation = useLocations as jest.Mock;
-const mockUserSession = useSessionUser as jest.Mock;
-const mockCreateAppointment = createAppointment as jest.Mock;
-const mockUseAppointmentService = useAppointmentService as jest.Mock;
-
-jest.mock('../hooks/useAppointmentService', () => ({
-  useAppointmentService: jest.fn(),
-  __esModule: true,
-  default: jest.fn(),
-}));
-
-jest.mock('@openmrs/esm-framework', () => ({
-  ...(jest.requireActual('@openmrs/esm-framework') as any),
-  useLocations: jest.fn(),
-  useSessionUser: jest.fn(),
-  showNotification: jest.fn(),
-  showToast: jest.fn(),
-}));
-
-jest.mock('./appointments.resource', () => ({
-  createAppointment: jest.fn(),
-}));
+const testProps = {
+  closeWorkspace: jest.fn(),
+  isTablet: false,
+  patientUuid: mockPatient.id,
+};
 
-describe('AppointmentForm', () => {
-  const renderAppointmentForm = () => {
-    mockUseLocation.mockReturnValue(mockLocations);
-    mockUserSession.mockReturnValue(mockSessionDataResponse.data);
-    mockUseAppointmentService.mockReturnValue({
-      status: 'resolved',
-      services: mockUseAppointmentServiceData,
-      error: null,
-    });
-    render(<AppointmentForm closeWorkspace={mockCloseWorkspace} patientUuid={mockPatient.id} isTablet={true} />);
+const mockCreateAppointment = createAppointment as jest.Mock;
+const mockDetach = detach as jest.Mock;
+const mockOpenmrsFetch = openmrsFetch as jest.Mock;
+const mockShowNotification = showNotification as jest.Mock;
+const mockShowToast = showToast as jest.Mock;
+
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+
+  return {
+    ...originalModule,
+    detach: jest.fn(),
+    useLocations: jest.fn().mockImplementation(() => mockLocations),
+    useSessionUser: jest.fn().mockImplementation(() => mockSessionDataResponse.data),
+    showToast: jest.fn(),
   };
+});
```

Comment:
Just for my understanding, why do we do this?

Reply:
I want to mock only a small subset of functions that the module provides. Here I'm saying I want to mock `detach` and `showToast` without actually overriding their implementations. This is because I want to be able to track calls to those functions so that I can assert that they were called a certain number of times and that they were called with certain arguments. For `useLocations` and `useSessionUser`, I want to override their implementations so that all the tests in my suite get access to the same locations and session data.